src/host/construction.jl

```diff
@@ -1,5 +1,14 @@
 # convenience and indirect construction
 
+# conversions from CPU arrays rely on constructors
+Base.convert(::Type{T}, a::AbstractArray) where {T<:AbstractGPUArray} = a isa T ? a : T(a)
```

Comment:
This looks like a potential for a lot of invalidations. Was that checked?

Reply:
No, I didn't check for that. We did used to have a `Base.convert(AT::Type{<: AbstractGPUArray}, iter)` method though, which is worse (right?), and I don't remember this convert method standing out when I had a quick look at invalidations a couple of weeks ago.

Reply:
Yeah, this should at least be better. Interesting.